### PR TITLE
Fix files init container

### LINKED ISSUE #270: BUG: init container -n flag required when outputting files format
When using the following flags to start the init container I would expect it to mount the secrets as files
```YAML
args: ["-e","/encrypted-secrets","-d","/decrypted-secret-files", "-f", "files"]
```

However, I get the following error:
> Failed to run init container: TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undef
ined

Adding the `-n` flag resolves the issue and correctly decrypts the files while ignoring the `-n` 
```YAML
args: ["-e","/encrypted-secrets","-d","/decrypted-secret-files", "-f", "files", "-n", "secrets.env"]
```

### PR DESCRIPTION
**Pull request by omerlh**
close #270

### CODE REVIEW

File: init-container/index.js [hunk 1]
```diff
@@ -93,10 +93,27 @@ async function innerRun() {
         }
     }
 
+    if (program.decryptedPath == null || program.decryptedPath == "")
```
Comment (shaikatz) [inline init-container/index.js]:
```suggestion
    if (!program.decryptedPath)
```

File: init-container/index.js [hunk 2]
```diff
@@ -93,10 +93,27 @@ async function innerRun() {
...
+    {
+      throw "decrypted path name wasn't provided altough it's mandataroy";
```
Comment (shaikatz) [inline init-container/index.js]:
```suggestion
      throw "decrypted path wasn't provided although it's mandatory";
```

File: init-container/index.js [hunk 3]
```diff
@@ -93,10 +93,27 @@ async function innerRun() {
...
+
     checkDirectorySync(program.decryptedPath);
-    
-    const outputFile = path.join(program.decryptedPath, program.decryptedFileName);
-    console.log(`Writing output format using ${program.outputFormat} format to file ${outputFile}`);
+
+    const outputFormat = program.outputFormat.toLowerCase();
+    let outputFile = "";
+
+    if (program.decryptedFileName == null || program.decryptedFileName == "")
```
Comment (shaikatz) [inline init-container/index.js]:
```suggestion
    if (!program.decryptedFileName) 
```

File: init-container/index.js [hunk 4]
```diff
@@ -93,10 +93,27 @@ async function innerRun() {
...
+
+    const outputFormat = program.outputFormat.toLowerCase();
+    let outputFile = "";
+
+    if (program.decryptedFileName == null || program.decryptedFileName == "") 
+    {
+      if(outputFormat != "files"){
+        throw "decrypted file name wasn't provided altough it's mandataroy";
+      }
+      console.log(`Writing output format using ${program.outputFormat} format to file ${program.decryptedPath}`);
```
Comment (shaikatz) [inline init-container/index.js]:
```suggestion
      console.log(`Writing output files to folder ${program.decryptedPath}`);
```

File: init-container/index.js [hunk 5]
```diff
@@ -93,10 +93,27 @@ async function innerRun() {
...
+    } else {
+      outputFile = path.join(program.decryptedPath, program.decryptedFileName);
+      console.log(`Writing output format using ${program.outputFormat} format to file ${outputFile}`);
```
Comment (shaikatz) [inline init-container/index.js]:
```suggestion
      console.log(`Writing output using ${program.outputFormat} format to file ${outputFile}`);
```

Comment (shaikatz) [inline init-container/index.js]:
```suggestion
      console.log(`Writing output using ${program.outputFormat} format to file ${outputFile}`);
```

File: src/decrypt-api/Startup.cs [hunk 6]
```diff
@@ -50,10 +50,7 @@ public Startup(IHostingEnvironment env)
             });
 
             services.AddSingleton<IKubernetes>(s => {
-                var config = string.IsNullOrEmpty(Configuration["Kubernetes:ProxyUrl"])
-                ? KubernetesClientConfiguration.BuildDefaultConfig()
-                : new KubernetesClientConfiguration { Host = Configuration["Kubernetes:ProxyUrl"] };
-                return new Kubernetes(config);
+                return new Kubernetes(KubernetesClientConfiguration.BuildDefaultConfig());
```
Comment (shaikatz) [inline src/decrypt-api/Startup.cs]: Isn't relevant to this PR

File: init-container/index.js (hunk 3 above)
Comment (omerlh) [reply init-container/index.js]: I remembered it's not the same as null or empty, but maybe I was wrong...

File: src/decrypt-api/Startup.cs (hunk 6 above)
Comment (omerlh) [reply src/decrypt-api/Startup.cs]: yep I needed it for local testing of the init container... so I thought I'll add it

Comment (shaikatz) [reply src/decrypt-api/Startup.cs]: Isn't the ProxyUrl is required for testing?

Comment (omerlh) [reply src/decrypt-api/Startup.cs]: This replaces it - like the CRD - use either configured `KUBECONFIG` or in cluster config in one line. This was added in recent versions of the SDK...

Comment (omerlh) [reply src/decrypt-api/Startup.cs]: NM it breaks the build - will fix it in a latter point.

**Pull request merged**